### PR TITLE
Print the effective skaffold.yaml configuration

### PR DESCRIPTION
**Pull request by dgageot**
Example on `examples/nodejs`:

```
$ skaffold diagnose --only-config -p dev
apiVersion: skaffold/v2beta3
kind: Config
build:
  artifacts:
  - image: node-example
    context: backend
    sync:
      manual:
      - src: src/**/*.js
        dest: .
    docker:
      dockerfile: Dockerfile
      buildArgs:
        ENV: development
  tagPolicy:
    gitCommit: {}
  local:
    concurrency: 1
deploy:
  kubectl:
    manifests:
    - k8s/*.yaml
```

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

**Comment by dgageot** [comment]
cc @matthewmichihara

**Comment by balopat** [comment]
Love it, I wanted to open a bug for this :D - this will be super useful for gitops workflows as well, where we can "hydrate" the skaffold config into a separate repo for audit purposes!

File: cmd/skaffold/app/cmd/diagnose.go [hunk 1]
```diff
@@ -30,25 +31,37 @@ import (
...
+
 // NewCmdDiagnose describes the CLI command to diagnose skaffold.
 func NewCmdDiagnose() *cobra.Command {
 	return NewCmd("diagnose").
 		WithDescription("Run a diagnostic on Skaffold").
+		WithExample("Search for configuration issues and print the effective configuration", "diagnose").
+		WithExample("Print the effective skaffold.yaml configuration for given profile", "diagnose --only-yaml --profile PROFILE").
 		WithCommonFlags().
+		WithFlags(func(f *pflag.FlagSet) {
+			f.BoolVar(&onlyConfig, "only-config", false, "Only prints the effective skaffold.yaml configuration")
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/diagnose.go:46]: `--only-config` reads oddly.  `--config-only` or `--yaml-only`?

File: cmd/skaffold/app/cmd/diagnose.go [hunk 2]
```diff
@@ -30,25 +31,37 @@ import (
...
+		}).
 		NoArgs(doDiagnose)
 }
 
 func doDiagnose(ctx context.Context, out io.Writer) error {
 	return withRunner(ctx, func(r runner.Runner, config *latest.SkaffoldConfig) error {
-		fmt.Fprintln(out, "Skaffold version:", version.Get().GitCommit)
-		fmt.Fprintln(out, "Configuration version:", config.APIVersion)
-		fmt.Fprintln(out, "Number of artifacts:", len(config.Build.Artifacts))
+		if !onlyConfig {
```
Comment (briandealwis) [inline cmd/skaffold/app/cmd/diagnose.go:53]: Since YAML supports comments, we could output this as comments?

**Pull request merged**